src/main/java/org/terasology/logic/inventory/StartingInventorySystem.java

```diff
@@ -0,0 +1,177 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.inventory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.registry.In;
+import org.terasology.world.block.BlockManager;
+import org.terasology.world.block.family.BlockFamily;
+import org.terasology.world.block.items.BlockItemFactory;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class StartingInventorySystem extends BaseComponentSystem {
+
+    private static final Logger logger = LoggerFactory.getLogger(StartingInventorySystem.class);
+
+    @In
+    BlockManager blockManager;
+
+    @In
+    InventoryManager inventoryManager;
+
+    @In
+    EntityManager entityManager;
+
+    @In
+    PrefabManager prefabManager;
+
+    BlockItemFactory blockFactory;
+
+    @Override
+    public void initialise() {
+        blockFactory = new BlockItemFactory(entityManager);
+    }
+
+    @ReceiveEvent(components = {InventoryComponent.class})
+    public void onStartingInventory(OnPlayerSpawnedEvent event,
+                                    EntityRef entityRef,
+                                    StartingInventoryComponent startingInventoryComponent) {
+        InventoryComponent inventoryComponent = entityRef.getComponent(InventoryComponent.class);
+        if (entityRef.getParentPrefab() != null) {
+            logger.info("Adding starting inventory to {}, entity has {} slots",
+                    entityRef.getParentPrefab().getName(), inventoryComponent.itemSlots.size());
+        }
+        for (StartingInventoryComponent.InventoryItem item : startingInventoryComponent.items) {
+            addToInventory(entityRef, item, inventoryComponent);
+        }
+        entityRef.removeComponent(StartingInventoryComponent.class);
+    }
+
+    private boolean addToInventory(EntityRef entityRef,
+                                   StartingInventoryComponent.InventoryItem item,
+                                   InventoryComponent inventoryComponent) {
+        String uri = item.uri;
+        int quantity = item.quantity;
+        if (uri == null) {
+            logger.warn("Improperly specified starting inventory item: Uri is null");
```

Comment:
Could we log some more info on the item? Without an additional hint the log message does not help much (despite of finding the code location...)

Reply:
I was thinking that the player might be able to infer what went wrong by the previous log messages. e.g. if the first item succeeded then they saw this message they could gather that the problem is the second item. I'm not really sure what else we could add to the message really.